R/elpd.R

```diff
@@ -0,0 +1,67 @@
+#' Generic (expected) log-predictive density
+#'
+#' The `elpd()` methods for arrays, and matrices can compute the expected log pointwise predictive density for a new dataset or the log pointwise predictive density of the observed data (an overestimate of the elpd).
+#'
+#' @export
+#' @inheritParams loo
+#'
+#' @details The `elpd()` function is an S3 generic and methods are provided for
+#'   3-D pointwise log-likelihood arrays, pointwise log-likelihood matrices.
+#'
+#'
+#' @examples
+#' ### Array methods for calculating the lpd of the observed data (using example objects included with loo package)
+#' LLarr <- example_loglik_array()
+#' elpd(LLarr)
+#'
+#'
+#'
+elpd <- function(x, ...) {
+  UseMethod("elpd")
+}
+
+#' @export
+#' @templateVar fn elpd
+#' @template array
+#'
+elpd.array <- function(x, ...) {
+
+  ll <- llarray_to_matrix(x)
+  pointwise <- pointwise_elpd_calcs(ll)
+  elpd_object(pointwise, dim(ll))
```

Comment:
After converting the array to a matrix I think you can just call the `elpd.matrix()` method. So: 

```suggestion
  ll <- llarray_to_matrix(x)
  elpd.matrix(ll)
```